modules/openapi-generator/src/main/resources/Java/libraries/webclient/api.mustache

```diff
@@ -127,6 +128,72 @@ public class {{classname}} {
+        {{/queryParams}}
+        {{/hasQueryParams}}
+        {{#hasHeaderParams}}
+
+        {{#headerParams}}
+        if ({{paramName}} != null)
+        headerParams.add("{{baseName}}", apiClient.parameterToString({{paramName}}));{{^-last}}
+        {{/-last}}
+        {{/headerParams}}
+        {{/hasHeaderParams}}
+        {{#hasCookieParams}}
+
+        {{#cookieParams}}
+        cookieParams.putAll(apiClient.parameterToMultiValueMap({{#collectionFormat}}ApiClient.CollectionFormat.valueOf("{{{collectionFormat}}}".toUpperCase(Locale.ROOT)){{/collectionFormat}}{{^collectionFormat}}null{{/collectionFormat}}, "{{baseName}}", {{paramName}}));
+        {{/cookieParams}}
+        {{/hasCookieParams}}
+        {{#hasFormParams}}
+
+        {{#formParams}}
+        if ({{paramName}} != null)
+            formParams.add{{#collectionFormat}}All{{/collectionFormat}}("{{baseName}}", {{#isFile}}{{^collectionFormat}}{{#useAbstractionForFiles}}{{paramName}}{{/useAbstractionForFiles}}{{^useAbstractionForFiles}}new FileSystemResource({{paramName}}){{/useAbstractionForFiles}}{{/collectionFormat}}{{/isFile}}{{#isFile}}{{#collectionFormat}}{{paramName}}.stream(){{^useAbstractionForFiles}}.map(FileSystemResource::new){{/useAbstractionForFiles}}.collect(Collectors.toList()){{/collectionFormat}}{{/isFile}}{{^isFile}}{{paramName}}{{/isFile}});
+        {{/formParams}}
+        {{/hasFormParams}}
+
+        final String[] localVarAccepts = { {{#hasProduces}}
+            {{#produces}}"{{{mediaType}}}"{{^-last}}, {{/-last}}{{/produces}}
+        {{/hasProduces}}};
+        final List<MediaType> localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);
+        final String[] localVarContentTypes = { {{#hasConsumes}}
+            {{#consumes}}"{{{mediaType}}}"{{^-last}}, {{/-last}}{{/consumes}}
+        {{/hasConsumes}}};
+        final MediaType localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);
+
+        String[] localVarAuthNames = new String[] { {{#authMethods}}"{{name}}"{{^-last}}, {{/-last}}{{/authMethods}} };
+
+        {{#returnType}}ParameterizedTypeReference<{{#isArray}}{{{returnBaseType}}}{{/isArray}}{{^isArray}}{{{returnType}}}{{/isArray}}> localVarReturnType = new ParameterizedTypeReference<{{#isArray}}{{{returnBaseType}}}{{/isArray}}{{^isArray}}{{{returnType}}}{{/isArray}}>() {};{{/returnType}}{{^returnType}}ParameterizedTypeReference<Void> localVarReturnType = new ParameterizedTypeReference<Void>() {};{{/returnType}}
+        return apiClient.{{#isArray}}invokeFluxAPIWithHttpInfo{{/isArray}}{{^isArray}}invokeAPIWithHttpInfo{{/isArray}}("{{{path}}}", HttpMethod.{{httpMethod}}, pathParams, queryParams, postBody, headerParams, cookieParams, formParams, localVarAccept, localVarContentType, localVarAuthNames, localVarReturnType);
+    }
```

Comment:
@UnleashSpirit Thanks for adding WithHttpInfo support. Should the original method (without http info) be updated to call this new method instead and just return the response (if not void) ? I think that's how other java clients provides both methods with or without `HttpInfo`

Ref: https://github.com/OpenAPITools/openapi-generator/pull/9327/files#diff-8f28a383fab93a6c7052a8ec90e18f3a50dc62b45cd516d17cdb92a3a07a096aR66

Reply:
Agree, but in fact you can't retrieve body from `Mono<ResponseEntity<T>>` without using `.block()`
Nethier get ResponseEntity once `requestBuilder.retrieve().bodyToMono(returnType)` is done

May be can do a private method in <Method>Api (api.pmustache) to avoid duplicate the creation request ?

Reply:
Something like 
```java
private ResponseSpec XXXCreateMonoRequest(...) {
	[actual request creation]
	return apiClient.invokeAPI(...);
}

public Mono<DTO> XXX(...) throws WebClientResponseException {
	ParameterizedTypeReference<DTO> localVarReturnType = new ParameterizedTypeReference<>() {};
	return XXXCreateMonoRequest(...).bodyToMono(localVarReturnType);
}

public Mono<ResponseEntity<DTO>> XXXWithHttpInfo(...) throws WebClientResponseException {
	ParameterizedTypeReference<DTO> localVarReturnType = new ParameterizedTypeReference<>() {};
	return XXXCreateMonoRequest(...).toEntity(localVarReturnType);
}
```

With ApiClient
```java
public <T> ResponseSpec invokeAPISpec(String path, HttpMethod method, Map<String, Object> pathParams, MultiValueMap<String, String> queryParams, Object body, HttpHeaders headerParams, MultiValueMap<String, String> cookieParams, MultiValueMap<String, Object> formParams, List<MediaType> accept, MediaType contentType, String[] authNames, ParameterizedTypeReference<T> returnType) throws RestClientException {
	final WebClient.RequestBodySpec requestBuilder = prepareRequest(path, method, pathParams, queryParams, body, headerParams, cookieParams, formParams, accept, contentType, authNames);
	return requestBuilder.retrieve();
}
```

And the same for `Flux`
??